tchannel-core/src/test/java/com/uber/tchannel/handlers/TestThriftAsyncRequestHandler.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.uber.tchannel.handlers;
+
+import com.uber.jaeger.Tracer;
+import com.uber.jaeger.reporters.InMemoryReporter;
+import com.uber.jaeger.samplers.ConstSampler;
+import com.uber.jaeger.samplers.Sampler;
+import com.uber.tchannel.tracing.TracingContext;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.uber.tchannel.BaseTest;
+import com.uber.tchannel.api.handlers.ThriftAsyncRequestHandler;
+import com.uber.tchannel.api.TChannel;
+import com.uber.tchannel.api.TFuture;
+import com.uber.tchannel.api.SubChannel;
+import com.uber.tchannel.messages.ThriftRequest;
+import com.uber.tchannel.messages.ThriftResponse;
+import com.uber.tchannel.messages.generated.Example;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.net.InetAddress;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class TestThriftAsyncRequestHandler extends BaseTest {
+
+    private static Tracer tracer;
+    private static InMemoryReporter reporter;
+    private static TracingContext tracingContext;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        reporter = new InMemoryReporter();
```

Comment:
This test doesn't have any assertions on the tracing functionality. We have a  [TracingPropagationTest](https://github.com/uber/tchannel-java/blob/5f18d0645750edf84168b3974bf8f804fb923039/tchannel-core/src/test/java/com/uber/tchannel/tracing/TracingPropagationTest.java) which would be a better place to add tests for tracing. Then, this test can simply verify that the handler behaves as expected.